### ISSUE TITLE
pre-compilation generating warnings

### ISSUE THREAD

**Issue by ExpandingMan** [JuliaLang/Compat.jl#648]
```
WARNING: eval into closed module Base:
Expr(:block, #= Symbol("/home/savastio/.julia/packages/Compat/hsya0/src/Compat.jl"):1891 =#, Expr(:call, :range, Expr(:parameters, Expr(:..., :kwargs)), :start, :stop) = Expr(:block, #= Symbol("/home/savastio/.julia/packages/Compat/hsya0/src/Compat.jl"):1891 =#, Expr(:call, :range, Expr(:parameters, Expr(:kw, :stop, :stop), Expr(:..., :kwargs)), :start)))
  ** incremental compilation may be broken for this module **

WARNING: Method definition range(Any, Any) in module Base at range.jl:90 overwritten at /home/savastio/.julia/packages/Compat/hsya0/src/Compat.jl:1891.
WARNING: Method definition #range(Any, typeof(Base.range), Any, Any) in module Base overwritten.
```
Is this a Compat issue?

**Comment by asinghvi17** [comment]
I have been having the same issue, on Julia v1.1, MacOS v10.4.1, Compat v1.5.0

**Comment by fredrikekre** [comment]
Likely caused by #642, cc @martinholters. Maybe we can just do the version check at runtime instead?

**Comment by ProV1denCEX** [comment]
Same issue. I think giving a depreciate or warning is appropriate. Maybe we need more time for package developers to do some changes.

**Comment by c42f** [comment]
I just ran into this. I silenced it for now with `pkg> pin Compat@1.4`. I'll probably wait until the warnings are removed from `Compat` before unpinning.

**Comment by jheinen** [comment]
> I just ran into this. I silenced it for now with `pkg> pin Compat@1.4`. I'll probably wait until the warnings are removed from `Compat` before unpinning.

Same for me - had to pin `Compat`.

**Comment by martinholters** [comment]
> Likely caused by #642, cc @martinholters. Maybe we can just do the version check at runtime instead?

Yes, that is quite surely the cause. But the problem is that we need to overwrite a Base method even on current Julia, so I don't see how doing the version check at run time could help. But please open a PR if you have an idea. I won't be able to do anything here today but it surely would be nice to have this fixed ASAP.

**Comment by fredrikekre** [comment]
Maybe we can be smart and restrict the arguments, and don't overwrite the `::Any, ::Any` method in base? Should be fine to restrict to `::Number` for example.

**Comment by oxinabox** [comment]
this is really annoying because breaking precomplilation breaks IJulia and other things that do not like it. (Something about precompilation kills certain packages)

**Comment by ExpandingMan** [comment]
Thanks for the quick fix @fredrikekre !
